iocore/eventsystem/I_Action.h

```diff
@@ -25,6 +25,10 @@
 #ifndef _I_Action_h_
 #define _I_Action_h_
 
+#define ACTION_RESULT_NONE ((Action *)(0x01))
+#define ACTION_RESULT_DONE ((Action *)(0x02))
+#define ACTION_IO_ERROR ((Action *)(0x04))
```

Comment:
Is there any significance to changing the numbering for these constants?  I assume they aren't persistent, so it shouldn't make a difference.  Are you trying to make a bitmask?

Reply:
I'm really just trying to simplify the macros. Before it was setting it to values like 1,3,5 and 7. I'm not sure it has to be a bitmask, but that was simple to do incase it is used like that somewhere. It kinda looked like the old macros might have intended that, but weren't achieving it.

Reply:
This should be `static constexpr` values inside the `Action` class. E.g.
```
static constexpr self_type * RESULT_NONE {reinterpret_cast<self_type*>(0x01)};
```

Reply:
I understand the previous macros, and that's not a bad idea. The goal is to make the pointers odd numbers, which are *always* invalid pointers. That makes them easy to distinguish from actual pointers. Note the macro shifted left and added on, i.e. opened bit 0 and set it, forcing the result to be odd.